src/Engine.cpp

```diff
@@ -602,6 +603,40 @@ bool Engine::entity_clicked(
     return res;
 }
 
+bool Engine::entity_unclick_(backend::EntityKind new_entity_clicked_kind)
+{
+    bool res = false;
+
+    // Generic part
+    // In case a real entity is clicked, deselect
+    if (last_entity_clicked_.is_valid_and_unique())
+    {
+        res = res || update_entity_generic(last_entity_clicked_, last_entity_clicked_kind_, true, false);
+    }
+
+    // Logical Physical part
+    // If new entity clicked is logical or physical, unselect the logical/physical one
+    if ((new_entity_clicked_kind == backend::EntityKind::HOST) ||
+            (new_entity_clicked_kind == backend::EntityKind::USER) ||
+            (new_entity_clicked_kind == backend::EntityKind::PROCESS) ||
+            (new_entity_clicked_kind == backend::EntityKind::DOMAIN) ||
+            (new_entity_clicked_kind == backend::EntityKind::TOPIC) ||
+            (new_entity_clicked_kind == backend::EntityKind::INVALID)) // In case invalid new kind, it means reset all
+    {
+        if (last_physical_logical_entity_clicked_.is_valid_and_unique())
+        {
+            res = res ||
+                update_entity_generic(
+                    last_physical_logical_entity_clicked_,
+                    last_physical_logical_entity_clicked_kind_,
+                    true,
+                    false);
+        }
+    }
+
+    return res;
+}
```

Comment:
When a participant is selected, the topic/domain/host/process to whom it belongs becomes unclicked.

Reply:
```suggestion
bool Engine::entity_unclick_(backend::EntityKind new_entity_clicked_kind)
{
    bool res = false;

    // Logical Physical part
    // If new entity clicked is logical or physical, unselect the logical/physical one
    if ((new_entity_clicked_kind == backend::EntityKind::HOST) ||
            (new_entity_clicked_kind == backend::EntityKind::USER) ||
            (new_entity_clicked_kind == backend::EntityKind::PROCESS) ||
            (new_entity_clicked_kind == backend::EntityKind::DOMAIN) ||
            (new_entity_clicked_kind == backend::EntityKind::TOPIC) ||
            (new_entity_clicked_kind == backend::EntityKind::INVALID)) // In case invalid new kind, it means reset all
    {
        if (last_physical_logical_entity_clicked_.is_valid_and_unique())
        {
            res = res ||
                update_entity_generic(
                    last_physical_logical_entity_clicked_,
                    last_physical_logical_entity_clicked_kind_,
                    true,
                    false);
        }
    }

    // Generic part
    // In case a real entity is clicked, deselect the last entity and hold the logical/physical one clicked
    else if ((last_entity_clicked_kind_ == backend::EntityKind::PARTICIPANT) ||
                (last_entity_clicked_kind_ == backend::EntityKind::DATAREADER) ||
                (last_entity_clicked_kind_ == backend::EntityKind::DATAWRITER) ||
                (last_entity_clicked_kind_ == backend::EntityKind::LOCATOR))
    {
        if (last_entity_clicked_.is_valid_and_unique())
        {
            res = res || update_entity_generic(last_entity_clicked_, last_entity_clicked_kind_, true, false);
        }
    }

    return res;
}
```